client/my-sites/site-settings/redirect-to/index.jsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import page from 'page';
+import { flowRight } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import { isSiteAutomatedTransfer } from 'state/selectors';
+
+const redirectNonJetpack = redirectRoute => WrappedComponent => {
+	class RedirectNonJetpack extends Component {
+		static propTypes = {
+			redirectRoute: PropTypes.string,
+			// Connected props
+			siteIsAtomic: PropTypes.bool,
+			siteIsJetpack: PropTypes.bool,
+			siteSlug: PropTypes.string
+		}
+
+		componentDidMount() {
+			this.redirectIfNoAccess();
+		}
+
+		componentDidUpdate() {
+			this.redirectIfNoAccess();
+		}
+
+		redirectIfNoAccess() {
+			if ( this.props.siteIsJetpack === false || this.props.siteIsAtomic ) {
+				this.redirectToDiffSite();
+			}
+		}
+
+		redirectToDiffSite = () => {
```

Comment:
I'm open to suggestions here, but I still don't think this name illustrates well enough what we're doing. Maybe just name it `redirect`? We definitely don't want to put `site` in the name, because we don't necessarily have to contain a site slug or ID in the redirect route.